superset/commands/chart/update.py

```diff
@@ -106,12 +107,12 @@ def validate(self) -> None:
 
         # validate tags
         try:
-            validate_tags(ObjectType.chart, self._model.tags, tag_ids)
+            validate_tags(ObjectType.CHART, self._model.tags, tag_ids)
         except ValidationError as ex:
             exceptions.append(ex)
 
         # Validate/Populate datasource
-        if datasource_id is not None:
+        if datasource_id is not None and datasource_type is not None:
```

Comment:
bycatch, a more recent linter would have likely started complaining about this sooner or later.